File: internal/tier/features.go
package tier

// Feature identifies a gatable capability in PayFlux.
// All access checks must use CanAccess(feature), not inline tier comparisons.
type Feature string

const (
	FeatureBasicRiskScore    Feature = "basic_risk_score"
	FeatureSlopeModeling     Feature = "slope_modeling"
	FeatureAcceleration      Feature = "acceleration_modeling"
	FeatureInstabilityIndex  Feature = "instability_index"
	FeatureReserveProjection Feature = "reserve_projection"
	FeatureConfidenceBands   Feature = "confidence_bands"
	FeatureSystemShockBlend  Feature = "system_shock_blend"
	FeatureAlertRouting      Feature = "alert_routing"
	FeatureEvidenceExport    Feature = "evidence_export"
	FeatureBulkExport        Feature = "bulk_export"
	FeatureExtendedRetention Feature = "extended_retention"
	FeatureHighConcurrency   Feature = "high_concurrency"
)

File: apps/dashboard/src/lib/tier/features.ts
/**
 * Feature identifies a gatable capability in PayFlux.
 * All access checks must use canAccess(feature), not inline tier comparisons.
 */
export type Feature =
  | "basic_risk_score"
  | "slope_modeling"
  | "acceleration_modeling"
  | "instability_index"
  | "reserve_projection"
  | "confidence_bands"
  | "system_shock_blend"
  | "alert_routing"
  | "evidence_export"
  | "bulk_export"
  | "extended_retention"
  | "high_concurrency";
